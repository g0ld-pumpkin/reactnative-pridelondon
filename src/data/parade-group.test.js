// @flow
import { generateCMSParadeGroup, sampleOne } from "./__test-data";
import decodeParadeGroup from "./parade-group";

describe("Parade Group", () => {
  describe("decoder", () => {
    it("correctly decodes valid CMS Parade Group", () => {
      const data: mixed = sampleOne(generateCMSParadeGroup);

      const decoded = decodeParadeGroup("en-GB")(data);
      expect(decoded.ok).toEqual(true);
      if (decoded.ok) {
        expect(decoded.value).toMatchSnapshot();
      }
    });

    it("correctly decodes valid CMS Parade Group with missing optional fields", () => {
      const data: mixed = {
        sys: {
          id: "jU1T9830fV3Bl3Z3fk7yktQp",
          contentType: {
            sys: {
              id: "paradeGroup"
            }
          },
          revision: 1
        },
        fields: {
          name: {
            "en-GB": "mm2T1WpGuu4k03kl04088fDh1B5V"
          }
        }
      };

      const decoded = decodeParadeGroup("en-GB")(data);
      expect(decoded.ok).toEqual(true);
      if (decoded.ok) {
        expect(decoded.value).toMatchSnapshot();
      }
    });

    it("fails if a property is missing", () => {
      const data: mixed = {
        fields: {},
        sys: {
          id: "3O3SZPgYl2MUEWu2MoK2oi",
          contentType: {
            sys: {
              id: "paradeGroup"
            }
          }
        }
      };

      const decoded = decodeParadeGroup("en-GB")(data);
      expect(decoded.ok).toEqual(false);
    });
  });
});
