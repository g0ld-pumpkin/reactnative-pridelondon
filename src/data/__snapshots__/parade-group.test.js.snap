// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parade Group decoder correctly decodes valid CMS Parade Group 1`] = `
Object {
  "contentType": "paradeGroup",
  "fields": Object {
    "facebookUrl": "https://red-badger.com/Ii3rT8J7D7oRAsi73e2G",
    "name": "BhabSLtI692l05Qliy85",
    "twitterUrl": "https://red-badger.com/24f3tJ9kLEIsCL0O8y881NG4pjYP2",
    "websiteUrl": null,
  },
  "id": "hG",
  "locale": "en-GB",
  "revision": 1,
}
`;

exports[`Parade Group decoder correctly decodes valid CMS Parade Group with missing optional fields 1`] = `
Object {
  "contentType": "paradeGroup",
  "fields": Object {
    "facebookUrl": null,
    "name": "mm2T1WpGuu4k03kl04088fDh1B5V",
    "twitterUrl": null,
    "websiteUrl": null,
  },
  "id": "jU1T9830fV3Bl3Z3fk7yktQp",
  "locale": "en-GB",
  "revision": 1,
}
`;
